x/ibc/20-transfer/module.go

```diff
@@ -1,31 +1,109 @@
 package transfer
 
 import (
+	"encoding/json"
+
 	"github.com/gorilla/mux"
 	"github.com/spf13/cobra"
 
+	abci "github.com/tendermint/tendermint/abci/types"
+
 	"github.com/cosmos/cosmos-sdk/client/context"
 	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
 	"github.com/cosmos/cosmos-sdk/x/ibc/20-transfer/client/cli"
-	"github.com/cosmos/cosmos-sdk/x/ibc/20-transfer/client/rest"
+	"github.com/cosmos/cosmos-sdk/x/ibc/20-transfer/types"
 )
 
+const (
+	ModuleName = types.SubModuleName
+)
+
+var (
+	_ module.AppModule      = AppModule{}
+	_ module.AppModuleBasic = AppModuleBasic{}
+)
+
+type AppModuleBasic struct{}
+
 // Name returns the IBC transfer ICS name
-func Name() string {
+func (AppModuleBasic) Name() string {
 	return SubModuleName
 }
 
-// RegisterRESTRoutes registers the REST routes for the IBC transfer
-func RegisterRESTRoutes(ctx context.CLIContext, rtr *mux.Router) {
-	rest.RegisterRoutes(ctx, rtr)
+func (AppModuleBasic) RegisterCodec(cdc *codec.Codec) {
+	RegisterCodec(cdc)
+}
+
+func (AppModuleBasic) DefaultGenesis() json.RawMessage {
+	return nil
 }
 
-// GetTxCmd returns the root tx command for the IBC transfer.
-func GetTxCmd(cdc *codec.Codec) *cobra.Command {
+func (AppModuleBasic) ValidateGenesis(bz json.RawMessage) error {
+	return nil
+}
+
+func (AppModuleBasic) RegisterRESTRoutes(ctx context.CLIContext, rtr *mux.Router) {
+	// TODO
+}
+
+func (AppModuleBasic) GetTxCmd(cdc *codec.Codec) *cobra.Command {
 	return cli.GetTxCmd(cdc)
 }
 
-// GetQueryCmd returns the root tx command for the IBC transfer.
-func GetQueryCmd(cdc *codec.Codec, queryRoute string) *cobra.Command {
-	return cli.GetQueryCmd(cdc, queryRoute)
+func (AppModuleBasic) GetQueryCmd(cdc *codec.Codec) *cobra.Command {
+	return cli.GetQueryCmd(cdc, QuerierRoute)
+}
+
+type AppModule struct {
+	AppModuleBasic
+	keeper Keeper
+}
+
+func NewAppModule(k Keeper) AppModule {
+	return AppModule{
+		keeper: k,
+	}
+}
+
+func (AppModule) RegisterInvariants(ir sdk.InvariantRegistry) {
+	// TODO
+}
+
+func (AppModule) Route() string {
+	return RouterKey
+}
+
+func (am AppModule) NewHandler() sdk.Handler {
+	return NewHandler(am.keeper)
+}
+
+func (AppModule) QuerierRoute() string {
+	return QuerierRoute
+}
+
+func (am AppModule) NewQuerierHandler() sdk.Querier {
+	//	return NewQuerier(am.keeper)
```

Comment:
uncomment?

Reply:
There is no querier for 20-transfer in the current codebase(but we need to add in the future)

Reply:
Is there an open issue for `x/ibc/20-transfer` queriers? Cause this seems like it would be blocking for GoZ.